sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/read/UnboundedSolaceReader.java

```diff
@@ -185,7 +185,14 @@ public Instant getWatermark() {
     if (getSessionService().getReceiver().isEOF()) {
       return BoundedWindow.TIMESTAMP_MAX_VALUE;
     }
-    return watermarkPolicy.getWatermark();
+    try {
+      return watermarkPolicy.getWatermark();
+    } catch (NullPointerException e) {
```

Comment:
It is generally a bad practice to catch NullPointerException.

Consider avoid it by if (watermarkPolicy.getWatermark() == null {
BoundedWindow.TIMESTAMP_MIN_VALUE; // with explanation
}
return watermarkPolicy.getWatermark();